### ISSUE TITLE
json 1.8.3: json/pure fails generating Fixnum

### ISSUE THREAD

**Issue by atombrenner** [flori/json#269]
The following snippet raises an exception when generating to_json with the pure parser.

```
require 'json/pure'
JSON('{"test": 5}').to_json
```

The C parser works like expected. If you change 5 to 5.1 it works again. Seems to be a wrong type check inside the pure generator.

Stacktrace:
`/var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:366:in `to_json': wrong argument type JSON::Pure::Generator::State (expected JSON/Generator/State) (TypeError)
    from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:366:in `block in json_transform'
    from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:359:in `each'
    from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:359:in `json_transform'
    from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:341:in `to_json'`

**Comment by atombrenner** [comment]
I looked into generator.rb and couldn't find any kind of type check. So I got back to my code and found out that it required 'aws-sdk' which required 'json'. So to reproduce the issue on irb you need to do this:
```
Tuxedo:~$ irb
irb(main):001:0> require 'json'
=> true
irb(main):002:0> require 'json/pure'
=> true
irb(main):003:0> JSON('{"test": 5}').to_json
TypeError: wrong argument type JSON::Pure::Generator::State (expected JSON/Generator/State)
	from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:366:in `to_json'
	from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:366:in `block in json_transform'
	from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:359:in `each'
	from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:359:in `json_transform'
	from /var/lib/gems/2.1.0/gems/json-1.8.3/lib/json/pure/generator.rb:341:in `to_json'
	from (irb):3
	from /usr/bin/irb:11:in `<main>'
irb(main):004:0> 5.to_json
TypeError: wrong argument type JSON::Pure::Generator::State (expected JSON/Generator/State)
	from (irb):4:in `to_json'
	from (irb):4
	from /usr/bin/irb:11:in `<main>'
irb(main):005:0> puts 5.1.to_json
5.1
=> nil

```

**Comment by marcandre** [comment]
Sorry for the late feedback.

This is fixed in 2.3.0 and can be closed.
